src/compile/cache/tests.rs

```diff
@@ -28,20 +25,77 @@ fn read_cache_with_valid_input() {
 fn read_non_existing_cache_file() {
 	let current_dir = std::env::current_dir().unwrap();
 	let cache_path = current_dir.join("test_cache_files").join("non_existing_cache.json");
-	let result = read_cache_file(&cache_path);
-	assert_matches!(result, Err(CacheError::FileNotFoundError(_)));
+	let result = CompileCacheItem::read(&cache_path);
+	assert_matches!(result, Err(CacheError::ReadFile(_, _)));
 }
 
 #[test]
 fn read_existing_cache_with_incorrect_field() {
 	let current_dir = std::env::current_dir().unwrap();
 	let cache_path = current_dir.join("test_cache_files").join("test_invalid_structure.json");
-	let result = read_cache_file(&cache_path);
-	assert_matches!(result, Err(CacheError::DeserializeError(_)));
+	let result = CompileCacheItem::read(&cache_path);
+	assert_matches!(result, Err(CacheError::DeserializeError(_, _)));
 }
 
 #[test]
-fn get_cache_path_for_valid_contract_path() {
+fn write_cache_with_valid_input() -> Result<(), CacheError> {
+	let cache_dir = std::env::temp_dir().join("cairo_foundry_test");
+
+	fs::create_dir_all(&cache_dir)
+		.map_err(|e| CacheError::DirCreation(cache_dir.as_path().display().to_string(), e))?;
+
+	let cache_path = cache_dir.join("write_cache_with_valid_input.json");
+
+	let cache = CompileCacheItem {
+		program_json: "".into(),
+		hash: 10,
+	};
+
+	CompileCacheItem::write(&cache, &cache_path)?;
+
+	let found_cache = CompileCacheItem::read(&cache_path)?;
+
+	assert_eq!(found_cache, cache);
+
+	Ok(())
+}
+
+#[test]
+fn update_cache_with_valid_input() -> Result<(), CacheError> {
+	let cache_dir = std::env::temp_dir().join("cairo_foundry_test");
+
+	fs::create_dir_all(&cache_dir)
+		.map_err(|e| CacheError::DirCreation(cache_dir.as_path().display().to_string(), e))?;
+
+	let cache_path = cache_dir.join("update_cache_with_valid_input.json");
+
+	let cache = CompileCacheItem {
```

Comment:
rename the variable to `expected_cache` for more readability.
Same for others instances of this behaviour